tools/build-openssl.sh

```diff
@@ -23,7 +23,7 @@ if ! grep -q "^VERSION=${OPENSSL_VERSION}$" build-openssl/Makefile ; then
     rm -rf build-openssl
     mkdir -p build-openssl
     pushd build-openssl
-    curl -s -l https://www.openssl.org/source/openssl-${OPENSSL_VERSION}.tar.gz | \
+    curl -s -l https://www.openssl.org/source/old/1.0.2/openssl-${OPENSSL_VERSION}.tar.gz | \
```

Comment:
Lessons learned from librdkafka; do `curl -fL <URL>`, otherwise it will not fail on 404.